src/SignaturePad.Shared/SignaturePadCanvasView.cs

```diff
@@ -398,11 +408,42 @@ private bool GetImageConstructionArguments (ImageConstructionSettings settings,
 			}
 
 			var sizeOrScale = settings.DesiredSizeOrScale.Value;
+#if NET471
+			var viewSize = this.RenderSize;
+#else
 			var viewSize = this.GetSize ();
+#endif
 
 			imageSize = sizeOrScale.GetSize ((float)viewSize.Width, (float)viewSize.Height);
 			scale = sizeOrScale.GetScale ((float)imageSize.Width, (float)imageSize.Height);
 
+#if NET471
+			if (settings.ShouldCrop == true)
+			{
+				signatureBounds = GetSignatureBounds (settings.Padding.Value);
+
+				if (sizeOrScale.Type == SizeOrScaleType.Size)
+				{
+					// if a specific size was set, scale to that
+					var scaleX = imageSize.Width / (float)signatureBounds.Width;
+					var scaleY = imageSize.Height / (float)signatureBounds.Height;
+					if (sizeOrScale.KeepAspectRatio)
+					{
+						scaleX = scaleY = Math.Min ((float)scaleX, (float)scaleY);
+					}
+					scale = new NativeSize ((int)scaleX, (int)scaleY);
+				}
+				else if (sizeOrScale.Type == SizeOrScaleType.Scale)
+				{
+					imageSize.Width = (int)(signatureBounds.Width * scale.Width);
+					imageSize.Height = (int)(signatureBounds.Height * scale.Height);
+				}
+			}
+			else
+			{
+				signatureBounds = new NativeRect (0, 0, (float)viewSize.Width, (float)viewSize.Height);
+			}
+#else
```

Comment:
Is there a reason for this copy?

Reply:
ah, I see. floats, ints and types... hopefully we can use the XxxF

Reply:
I've just updated review, and remove all redundant # where it possible

Reply:
viewSize Width is double so we can't avoid casting